src/lib.rs

```diff
@@ -103,18 +105,58 @@ pub fn collect_stars(config: Config) -> Result<(), Box<dyn error::Error>> {
     if let Some(ref token) = config.token {
         builder.set_authorization_token(token.to_owned());
     }
+    else {
+        println!("Authentication Warning: This is an unauthenticated request with a limit of 60 requests per hour. Re-run this program using an auth token by adding `--token <auth-token>` for an increased quota of 5000 requests per hour.")
+    }
 
     let client = builder.build()?;
 
     let mut stars: Vec<Star> = Vec::new();
 
     let mut next_link = config.url();
 
+    let total_rate_limit = "X-RateLimit-Limit";
+    let rate_limit_remaining = "X-RateLimit-Remaining";
+    let rate_limit_reset_time = "X-RateLimit-Reset";   
+    
+    let mut remaining: i32 = 0; 
+    let mut total: i32 = 0;
+    let mut timestamp_str = String::new();
     while next_link.is_some() {
         if let Some(link) = next_link {
             let mut res = client.get(&link).send()?;
-            next_link = extract_link_next(res.headers());
 
+            for header in res.headers().iter() {
```

Comment:
Yes, that's right. I've opened up issue #36 to implement the Header trait, and will change this code around once that is completed.

Reply:
It doesn't necessarily have to be part of this pull request, the current approach works just fine.

I'd actually encourage that we leave #36 as a later enhancement and finish getting this pull request merged in, since implementing custom `Header`s isn't critical to getting these warnings implemented.